### ISSUE TITLE
AWS.Logger.SeriLog ignores Scope in CloudWatch logs

### ISSUE THREAD

**Issue by prindacerk** [aws/aws-logging-dotnet#107]
I have setup Serilog in my project and enabled both File writing and AWS CloudWatch logging. In my file logs, I can see the Scope information (or ForContext if I used Serilog logger directly).

But when I look at the CloudWatch streams, I can see the message only. If it is an exception, it logs the exception details. But the type of log (Error, Info, Warning etc.), any context information passed along are missing.

Is there anyway to absorb those information onto CloudWatch? I have set 
```
"IncludeLogLevel": true,
"IncludeCategory": true,
"IncludeNewline": true,
"IncludeException": true,
"IncludeEventId": false,
"IncludeScopes": true,
```
under Logging and under Serilog in appsettings. RollingFile sink is using it but AWSSeriLog isn't.

Please advice.

**Comment by klaytaybai** [comment]
https://github.com/aws/aws-logging-dotnet/blob/8f18e64006c16059e987ccac055487c5e78f4c0f/src/AWS.Logger.SeriLog/AWSLoggerSeriLogExtension.cs#L83-L93

When you are creating your logging configuration, you need to use an ITextFormatter that includes LogLevel, such as
https://github.com/serilog/serilog/blob/dev/src/Serilog/Formatting/Json/JsonFormatter.cs


Please let me know if that helps.  Thanks

**Comment by prindacerk** [comment]
I'm loading Serilog configuration via appsettings and using ReadFrom.Configuration.

You can have a look at the test project I'm experimenting with here.
https://github.com/prindacerk/LoggerTest

Not sure where you want me to set the formatter since the configuration is loaded via appsettings.

**Comment by prindacerk** [comment]
Is there any feedback on this? Any instruction on how to configure the formatter while loading via configuration?

**Comment by ashishdhingra** [comment]
Hi @prindacerk,

Good afternoon.

I was going through the issue backlog and came across this guidance question.

You might want to create custom `ITextFormatter` implementation as in example below:
```C#
namespace SerilogTestCodeFromConfig
{
    public class AWSTextFormatter : ITextFormatter
    {
        public void Format(LogEvent logEvent, TextWriter output)
        {
            output.Write("Timestamp - {0} | Level - {1} | Message {2} {3}", logEvent.Timestamp, logEvent.Level, logEvent.MessageTemplate, output.NewLine);
            if (logEvent.Exception != null)
            {
                output.Write("Exception - {0}", logEvent.Exception);
            }
        }
    }
}
```

and then modify the `appsettings.json` to include `Args` with textFormatter class specified, in `WriteTo` element:
```JSON
"WriteTo": [
      {
        "Name": "AWSSeriLog",
        "Args": {
          "textFormatter": "SerilogTestCodeFromConfig.AWSTextFormatter, SerilogTestCodeFromConfig"
        }
      }
``` 

The above JSON assumes that the code is compiled into assembly named `SerilogTestCodeFromConfig`.

Please give it a try and let me know if it works. 

Thanks,
Ashish

**Comment by prindacerk** [comment]
I tried configuring the formatter as above.

But I am getting the following exception when it is trying to create a logger from configuration in the appsettings.
`Invalid cast from 'System.String' to 'Serilog.Formatting.ITextFormatter'.`

The error is happening in here.
https://github.com/prindacerk/LoggerTest/blob/50e32051435b16be55eb41dac926796e6e22c94a/LoggerTest/Program.cs#L30-L33

Please advise on what could be the issue?

**Comment by ashishdhingra** [comment]
Hi @prindacerk,

Looks like this problem is reported in the issue https://github.com/aws/aws-logging-dotnet/issues/58#issuecomment-535437573 as well. I will try to do some more research into it, or else I would convert this to `feature-request`.

Thanks,
Ashish

**Comment by prindacerk** [comment]
@ashishdhingra Not sure what was wrong before. Tried today and it seems to be working. The exception is not being thrown and the custom formatting works.

I was even able to use the default JsonFormatter which seems to include all the additional properties into Cloudwatch as the RollingFile. So it seems I can use the default options instead of using custom options.

I also found an alternative nuget package that is tied to Cloudwatch specifically. **"Serilog.Sinks.AwsCloudWatch"**
It has more control on the cloudwatch logging if anyone is interested. For simple implementation, this sink is sufficient.

I am closing this. But I would recommend having a proper Readme with maybe code sample as I have shared (maybe a generic one) would be helpful for others who struggle with my kind of problem.
